pkl-core/src/main/java/org/pkl/core/runtime/VmUtils.java

```diff
@@ -298,6 +298,9 @@ public static Object doReadMember(
           }
         }
       } else if (receiver instanceof VmListingOrMapping<?> vmListingOrMapping) {
+        if (owner != receiver && owner instanceof VmListingOrMapping<?> vmListingOrMappingOwner) {
+          ret = vmListingOrMappingOwner.typecastObjectMember(member, ret, callNode);
```

Comment:
Why the `ret =` if in the next line you reassign to `ret`?

Reply:
`ret` is also an argument to the function call in the line below; it doubles the cast. That said; @translatenix's suggestion should be sufficient.

Reply:
Tests pass even after removing the first `ret =`.